addons/stock/models/stock_move.py

```diff
@@ -695,10 +712,10 @@ def onchange_quantity(self):
 
     @api.onchange('product_id')
     def onchange_product_id(self):
-        product = self.product_id.with_context(lang=self.partner_id.lang or self.env.user.lang)
-        self.name = product.partner_ref
-        self.product_uom = product.uom_id.id
-        return {'domain': {'product_uom': [('category_id', '=', product.uom_id.category_id.id)]}}
+        product = self.product_id
+        if product:
```

Comment:
Before on original function there wasnt any control if product. Was there any error , do we need the control if product?

Reply:
There wasn't an error no, it would just set the `product_uom.category_id` domain to False.

I was just thinking if a product hasn't been set, then it isn't worth trying to set/update the `product_uom` field or the domain, as `product_id` is a required field.